### PR TITLE
Feature/Custom implementation of geocoding

### PR DESCRIPTION
**Pull request by mauriceackel**
**What changes does this PR introduce**

Added custom geocoding and autocomplete functionality and ditched Google Maps API.

### CODE REVIEW

File: src/services/GeoService.js [hunk 1]
```diff
@@ -1,55 +1,62 @@
...
-        return reject(status);
-    }
-  }));
+import fb from '../firebase';
+
+const numericRegex = /\d{3,}/;
+const stringRegex = /\D+/;
+
+function enrichEntry(entry) {
+  return { ...entry, description: `${entry.plz || ''}${entry.plz ? ' ' : ''}${entry.name}` };
```
Comment (jagaSto) [inline src/services/GeoService.js:7]:
```suggestion
  return { ...entry, description: `${entry.plz ? `${entry.plz} ` : ''}${entry.name}` };
```

Comment (jagaSto) [reply src/services/GeoService.js:7]: a bit more readable like this?

File: src/services/GeoService.js [hunk 2]
```diff
@@ -1,55 +1,62 @@
...
-        return resolve(results);
-      }
-      return reject(status);
+export async function getSuggestions(searchString) {
+  const plzMatch = searchString.match(numericRegex);
+  const locationMatch = searchString.match(stringRegex);
+  const location = locationMatch !== null ? locationMatch[0].trim() : '';
+
+  if (plzMatch !== null) {
+    const plz = plzMatch[0];
```
Comment (jagaSto) [inline src/services/GeoService.js:21]:
```suggestion
    const [ plz ] = plzMatch;
```

File: src/services/GeoService.js [hunk 3]
```diff
@@ -1,55 +1,62 @@
...
-  return new Promise((resolve, reject) => {
-    geocoder.geocode(request, (results, status) => {
-      if (status === 'OK') {
-        return resolve(results);
-      }
-      return reject(status);
+export async function getSuggestions(searchString) {
+  const plzMatch = searchString.match(numericRegex);
+  const locationMatch = searchString.match(stringRegex);
+  const location = locationMatch !== null ? locationMatch[0].trim() : '';
```
Comment (jagaSto) [inline src/services/GeoService.js:18]: could `locationMatch `ever be an empty array here? If yes, we should check, so `locationMatch[0].trim()` does not break

**Comment by jagaSto** [comment]
found issues on dev so far: 
* filtering with old entries that have a country does not work, e.g. for `http://localhost:3000/#/?address=4020%20Linz,%20%C3%96sterreich`
* input for lower case does not find any matches e.g., try `mannheim` or `berlin`

File: src/services/GeoService.js [hunk 4]
```diff
@@ -1,55 +1,62 @@
...
+        lon: average(aggregatedEntries.map((e) => Number.parseFloat(e.lon))),
+        description: key,
+      };
     });
-  });
+}
+
+export async function getGeodataForString(searchString) {
+  const result = await getSuggestions(searchString);
+  return result[0];
```
Comment (jagaSto) [inline src/services/GeoService.js:52]:
```suggestion
  return [result];
```

File: src/services/GeoService.js [hunk 5]
```diff
@@ -1,55 +1,62 @@
...
-        return resolve(results);
-      }
-      return reject(status);
+export async function getSuggestions(searchString) {
+  const plzMatch = searchString.match(numericRegex);
+  const locationMatch = searchString.match(stringRegex);
+  const location = locationMatch !== null ? locationMatch[0].trim() : '';
+
+  if (plzMatch !== null) {
+    const [ plz ] = plzMatch;
```
Comment (jagaSto) [inline src/services/GeoService.js:21]:
```suggestion
    const [plz] = plzMatch;
```

**Comment by mauriceackel** [comment]
WARNING! Fixed case-insensitivity, but this requires to perform a migration on the prod database. Do not merge before!

**Pull request merged**